pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -674,7 +670,14 @@ stdenv.mkDerivation {
   preFixup = lib.optionalString withEfi ''
     mv $out/lib/systemd/boot/efi $out/dont-strip-me
   '';
-  postFixup = lib.optionalString withEfi ''
+
+  postFixup = ''
+    # Wrap in the correct path for LUKS2 tokens. Must be after the fixup phase
+    # or the rpath cleanup removes the directories again.
+    for f in lib/systemd/systemd-cryptsetup bin/systemd-cryptenroll; do
+      wrapProgram $out/$f --prefix LD_LIBRARY_PATH : ${placeholder "out"}/lib/cryptsetup
```

Comment:
Why do we need to set `LD_LIBRARY_PATH`? I was under the assumption that `${systemd}/lib` being in rpath of all systemd binaries calling out libcryptsetup code is sufficient…

Reply:
No, `${systemd}/lib` is not the same thing as `${systemd}/lib/cryptsetup`

Reply:
So `patchelf --add-rpath`, and disable the rpath trimming?

Reply:
Agreed. `LD_LIBRARY_PATH` is not the solution.

Also, in `systemdMinimal`, these executables don't exist, so the build fails.